src/main/java/io/confluent/idesidecar/restapi/models/ConnectionSpec.java

```diff
@@ -483,21 +483,24 @@ public List<Error> validate() {
       "CyclomaticComplexity",
       "NPathComplexity"
   })
-  public List<Error> validateUpdate(ConnectionSpec newSpec) {
+  public List<Error> validateUpdate(ConnectionSpec newSpec, boolean isPatch) {
     var errors = new ArrayList<Error>();
 
     // Check required fields and immutability
-    if (newSpec.name == null || newSpec.name.isBlank()) {
-      checkRequired(errors, "name", "Connection name");
+    if (!isPatch) {
+      if (newSpec.name == null || newSpec.name.isBlank()) {
+        checkRequired(errors, "name", "Connection name");
+      }
+      if (newSpec.type == null) {
+        checkRequired(errors, "type", "Connection type");
+      }
     }
     if (newSpec.id == null || newSpec.id.isBlank()) {
       checkRequired(errors, "id", "Connection ID");
     } else if (!Objects.equals(newSpec.id, id)) {
       checkImmutable(errors, "id", "Connection ID");
     }
-    if (newSpec.type == null) {
-      checkRequired(errors, "type", "Connection type");
-    } else if (!Objects.equals(newSpec.type, type)) {
+    if (!Objects.equals(newSpec.type, type)) {
```

Comment:
Was this another change leftover from the previous implementation? Ideally we still want to check if the new spec declares type as null.

Reply:
I'm checking in the method itself now... thoughts on that?

Reply:
We'd need to change the method signature on `validateUpdate` to track whether it's a PUT or PATCH, which I've put back in the code. This is because we don't need 'name' with a PATCH if a user doesn't want to PATCH that field, but PUT requires the whole thing. Setting it as null is something we probably don't want for either, so I also updated the logic.